R/logger.R

```diff
@@ -67,6 +67,18 @@ fallback_namespace <- function(namespace) {
     namespace
 }
 
+#' Assure valid fun_name
+#' @param fun_name string referencing current function name
+#' @param fallback_name string with fallback name if `fun_name` is invalid
+#' @keywords internal
+fallback_fun_name <- function(fun_name, fallback_name) {
+    fun_name_base <- strsplit(fun_name, '_')[[1]][2]
+    if (is.na(fun_name_base) || !is.character(fun_name_base)) {
+        return(fallback_name)
+    }
+    fun_name
```

Comment:
I kept using the `fun_name` to minimize changes

Ideally the `fun_name_base` would be returned and the documentation/code for `log_config_setter()` would change to avoid repeated code (`fun_name_base <- strsplit(fun_name, '_')[[1]][2]`)